### PR TITLE
chore: Update entities dependency

### LINKED ISSUE #900: Update entities to 4.x
👋 Would be great to see the usage of `entities` here updated to 4.x -- the new major version seems to have pretty substantial improvements to resulting bundle sizes :)

### LINKED ISSUE #916: Module not found: Error: Package path ./lib/maps/entities.json is not exported from package
I was curious what the outcome of the following issue is https://github.com/fb55/entities/issues/783

I'm running into this with the latest version of webpack as well as latest version of this library. 
```
Module not found: Error: Package path ./lib/maps/entities.json is not exported from package /Users/blakeniemyjski/code/abc/node_modules/entities (see exports field in /Users/blakeniemyjski/code/abc/entities/package.json)
 @ ./node_modules/markdown-it/lib/common/utils.js 80:15-36
 ```

### LINKED ISSUE #958: Unable to resolve "entities/lib/maps/entities.json" from "../../node_modules/markdown-it/lib/common/entities.js"
iOS Bundling failed 1640ms
Unable to resolve "entities/lib/maps/entities.json" from "../../node_modules/markdown-it/lib/common/entities.js"

Screenshot 2023-09-20 at 2 22 25 PM

"expo": "~48.0.18",
"react": "18.2.0",
"react-native": "0.71.8",

markdown-it version 10

### PR DESCRIPTION
**Pull request by MattIPv4**
Resolves #900, by updating `entities` to the 4.x major version.

This should also resolve #916 and resolve #958 as those are a result of the loose version pin for `entities`, resulting in some folks using 4.x already.

This includes a slight change to the order of operations in `replaceEntityPattern`, ensuring that the custom digital entity handling happens first (including bailing out early if the logic determines it to be an invalid code point, as `decodeHTML` from `entities` would "blindly" decode it -- this seems to now more closely follow how `lib/rules_inline/entity.js` already operated). This change was needed to keep the `markdown-it fatal.txt line 17` spec passing.

### CODE REVIEW

**Comment by bradenmacdonald** [comment]
Hey @MattIPv4 thanks for this work - I was trying to improve on my app's bundle size and got here in my search to shrink markdown-it's size. All those entities are a big contributor. I think I found an issue with this though...

As I understand the big value of entities v4.4 vs 3 is the smaller bundle size. But the bundle size on this branch actually _increases_ from 101 KiB to 140 KiB. I think the reason is that before it was only using `entities.json` but now you're using `decodeHTML` which brings more code with it. I haven't investigated in detail though.

I did find that changing the import of `decodeHTML` to `require('entities/lib/decode.js').decodeHTML` brings the minified size down to 115 KiB, but that's still an increase over the status quo.

(I was using `npm run browserify && ls -lh dist/` to see the effects on the final bundle.)

**Comment by MattIPv4** [comment]
Fwiw, I hadn't actually compared the direct bundle size of markdown-it in isolation in change, I was looking at it as part of a bigger picture with many other dependencies that we use. Many of those dependencies also depended on entities, but were using v4. With markdown-it still using v3, this leads webpack to include a copy of entities per dependency that needs in (v3 for markdown-it and a copy of v4 for each dependency that uses v4). By having markdown-it also using v4, webpack can heavily dedupe suddenly and only bundle a single copy of entities v4, massively reducing the bundle size.

File: lib/common/utils.js [hunk 1]
```diff
@@ -93,6 +89,13 @@ function replaceEntityPattern(match, name) {
     if (isValidEntityCode(code)) {
       return fromCodePoint(code);
     }
+
+    return match;
+  }
+
+  decoded = decodeHTML(match);
```
[the remark this answers is not in the document]
Comment (MattIPv4) [reply lib/common/utils.js:96]: `decodeUTF8` does not exist, the FAQ you linked is for encoding...

[the remark this answers is not in the document]
Comment (pyhedgehog) [reply lib/common/utils.js:96]: Sorry -- missed with `escapeUTF8`...

**Comment by tankbusta** [comment]
Any chance this could get merged? This also fixes a bug I'm running into with Vue 3 + Vite 4.5.0 on Windows

```
6:55:51 PM [vite] error while updating dependencies:
Error: Build failed with 1 error:
node_modules/markdown-it/node_modules/entities/lib/maps/entities.json:1:9: ERROR: Expected ";" but found ":"
    at failureErrorWithLog (<path>\node_modules\esbuild\lib\main.js:1649:15)
    at <path>\node_modules\esbuild\lib\main.js:1058:25
    at <path>\node_modules\esbuild\lib\main.js:1525:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    ```

**Pull request merged**